week-10/20194201/hust-student/src/Server_SV/controllers/student.js

```diff
@@ -0,0 +1,121 @@
+const mongoose = require("mongoose");
+const multer = require("multer");
+const studentModel = require("../models/student");
+const studentController ={};
+studentController.addStudent = async(req, res, next) =>{
+    console.log(req.file);
+    const files = req.file;
+   let avatar;
+    if(!files){
+       avatar = req.body.avatar
+    }else{
+       avatar = req.file.path
+    }
+    const student = new studentModel({
+        _id: mongoose.Types.ObjectId(),
+        mssv: req.body.mssv,
+        name: req.body.name,
+        class: req.body.class,
+        course: req.body.course,
+        email: req.body.email,
+        phone: req.body.phone,
+        address: req.body.address,
+        avatar: avatar
+    });
+    try{
+       let newStudent =  await student.save()
+        return res.status(201).json({
+            success:true,
+            message:"New student created succesfully.",
+            data: newStudent
+        }); 
+    }catch(error){
+        console.log(error);
+            res.status(500).json({
+                success: false,
+                message: "Server error, try again.",
+                error: error.message
+            });
+    }
+}
+studentController.getAllStudent = async(req, res, next) =>{
+        try{
+            let result = await studentModel.find() 
+                            . select("_id mssv name class course email phone address avatar");
+            return  res.status(200).json(result);
+
+        }catch(error){
+            res.status(500).json({
+                success: false,
+                message: "Can't Save. Try Again.",
+                error: error.message  
+            });
+        }
+}
+studentController.getStudent = async(req, res, next) =>{
+    const [mssv, name, id] = [req.params.mssv,  req.params.name, req.params.id];
+    try{
+        let result = await studentModel.find({
+            $or:[
+                {_id: id},
+                {mssv: mssv},
+                {name: name},
+            ]
+        });
+        if(result == []){
+            return res.status(500).json({
+                success:false,
+                message:"Not found student"
+            });
+        }
+        res.status(200).json(result);
+    }catch(error){
+        res.status(500).json({
+            success: false,
+            message:"This student dose not exits",
+            error: error.message
+        });
+    }
+}
+studentController.updateStudent = async(req, res, next) =>{
+    const [idStudent, updates] = [req.params.id, req.body];
```

Comment:
Trong trường hợp này thì ko cần dùng contruction  tại 1 cái lấy từ params 1 cái lầy từ body mà
```suggestion
    const idStudent =  req.params.id;
    const updates = req.body;
```